### PR TITLE
add value/data/seq properties to message

### PR DESCRIPTION
**Pull request by swathipil**
related to: https://github.com/Azure/azure-sdk-for-python/pull/24259

### CODE REVIEW

File: uamqp/message.py [hunk 1]
```diff
@@ -235,6 +235,33 @@ def delivery_annotations(self):
...
+    def data(self):
+        if not self._message or not self._body:
+            return None
+        # pylint: disable=protected-access
+        if self._body.type == c_uamqp.MessageBodyType.DataType:
+            return self._body.data
+        return None
+
+    @property
+    def sequence(self):
```
Comment (swathipil) [inline uamqp/message.py:248]: @annatisch - currently this property is returning the same generator each time. After it's been used once, it returns None. Do you know if there's any way to have it return a new generator each time, like `get_data` does?

Comment (swathipil) [reply uamqp/message.py:248]: update: this is not being used up, and works when called multiple times.

File: uamqp/client.py [hunk 2]
```diff
@@ -386,7 +386,7 @@ def client_ready(self):
             return False
         return True
 
-    def do_work(self):
+    def do_work(self, **kwargs): # pylint: disable=unused-argument
```
Comment (annatisch) [inline uamqp/client.py:389]: This needs to be added to `ClientAsync.do_work_async`

Comment (swathipil) [reply uamqp/client.py:389]: updated

**Review by annatisch — APPROVED** [review]

**Comment by swathipil** [comment]
test run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1782646&view=results

File: uamqp/message.py [hunk 3]
```diff
@@ -250,7 +250,10 @@ def data(self):
 
     @property
     def sequence(self):
-        if not self._message or not self._body:
+        try:
+            if not self._message or not self._body:
```
Comment (swathipil) [inline uamqp/message.py:254]: BatchMessage doesn't have a _message attribute, so will go into except and return None.

**Pull request merged**